mysql-test/suite/rpl/t/rpl_show_slave_hosts.test

```diff
@@ -49,4 +49,170 @@ source include/wait_show_condition.inc;
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT
+SHOW SLAVE HOSTS;
+
+--echo # Test 2: Primary enabled semisync
+--echo # Note:   it should be the same as test 1
+--echo # -------------------------------------------
+set global rpl_semi_sync_master_enabled = 1;
+show variables like 'rpl_semi_sync_master_enabled';
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT
+SHOW SLAVE HOSTS;
+
+
+--echo # Test 3: Replica enabled semisync
+--echo # Note:   No new GTIDs expected empty result
+--echo # -------------------------------------------
+connection slave;
+stop slave;
+reset slave;
+set global rpl_semi_sync_slave_enabled = 1;
+start slave;
+show variables like 'rpl_semi_sync_slave_enabled';
+source include/wait_for_slave_io_to_start.inc;
+connection master;
+
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT
+SHOW REPLICA HOSTS;
+show variables like 'rpl_semi_sync_master_enabled';
+
+--echo # Test 4: Make new events - generate new GTID
+--echo # Note:   
+--echo # -------------------------------------------
+create table t(t int);
+insert into t values (1);
+
+show status like 'Rpl_semi_sync_master_status';
+show status like 'Rpl_semi_sync_master_no_tx';
+show status like 'Rpl_semi_sync_master_yes_tx';
+let $status_var= Rpl_semi_sync_master_clients;
+let $status_var_value= 1;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_clients';
+# Here we should get ACK
+--replace_result $SLAVE_MYPORT SLAVE_PORT $DEFAULT_MASTER_PORT DEFAULT_PORT
+SHOW REPLICA HOSTS;
+
+
+--echo # Test 5: Connect with slave2 (server_id=3)
+--echo # Note:   In this test semisync will be enabled and new events created
+--echo # -------------------------------------------
+connection slave2;
+stop slave;
+reset slave;
+set global rpl_semi_sync_slave_enabled = 1;
+start slave;
+show variables like 'rpl_semi_sync_slave_enabled';
+
+
+# Check primary
+connection master;
+
+# - Check number of semi-sync replicas - 2 are connected
+let $status_var= Rpl_semi_sync_master_clients;
+let $status_var_value= 2;
+source include/wait_for_status_var.inc;
+show status like 'Rpl_semi_sync_master_clients';
+
+# - Check `show replica hosts` on primary without events on server_id=3
+# - This should replicate previous transaction (but is not sending the ack?)
+--echo # Question 1: Replica not sending ACK here (not good, expected, should wait on something?)
+--echo # For waiting: here Rpl_semi_sync_master_yes_tx = 2 (number of transactions already ack-ed by server_id=1)
+--echo # Should be reset somehow for this particular slave (server_id=3) and wait that yes_tx be 2 ? Or other apporach?
+--echo # Note that primary doesn't report not acked transaction by replicas.
+--echo # Update: Looking into documentation - "semisync replication waits only on 1 replica to respond" ->
+--echo # So according to documentation it is ok, but in later example you will see that both replica have sent ACK.
```

Comment:
This is actually because the master is re-sending old transactions. Only new transactions on the master side will be `ACK`ed by the replica. See logic pertaining to `k_packet_flag_sync` for more details. This is a good test to keep to prove that is the case, can you re-word the test description to reflect that behavior? Then the next test case does `ACK`  a new transaction, which is good.